src/covariance.h

```diff
@@ -127,6 +127,19 @@ matrix<T> get_compound_symmetry_heterogeneous(const vector<T>& theta, int n_visi
   return get_heterogeneous_cov(sd_values, fun);
 }
 
+//Spatial Exponential Cholesky factor.
+template <class T>
+matrix<T> get_spatial_exponential(const vector<T>& theta, const matrix<T>& distance) {
+  T const_sd = exp(theta(0));
+  T logrho = exp(theta(1));
+  matrix<T> expdist = exp(-distance.array() *  logrho);
+  matrix<T> result = expdist * const_sd;
+  matrix<T> chol(distance.rows(), distance.cols());
```

Comment:
can you try without defining this first? I think you can just directly return the line 139 result.